### PR TITLE
Support for /v1/synthetics

### PR DESCRIPTION
**Pull request by btoueg**
Follow up on https://github.com/zorkian/go-datadog-api/pull/205

### CODE REVIEW

**Comment by btoueg** [comment]
Please ping me before merging, as we are about to bump the version of the API, so I might still add a few commits today

**Comment by rtyler** [comment]
Really looking forward to this coming in and unblocking Terraform support for Synthetics :clap:

**Comment by ojongerius** [comment]
@btoueg happy to merge when you are ready. Thanks for helping @dharada1 getting this over the finish line! 🙇

**Comment by irabinovitch** [comment]
@ojongerius Thanks! As you can see from a few recent PRs were generally looking to get more involved in the project.  Let us know if theres anyway we (the Datadog team) can help with on going development.

File: synthetics.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	Name   *string `json:"name,omitempty"`
+	Email  *string `json:"email,omitempty"`
+	Handle *string `json:"handle,omitempty"`
+}
+
+type SyntheticsDevice struct {
+	Id          *string `json:"id"`
+	Name        *string `json:"name"`
+	Height      *int    `json:"height"`
+	Width       *string `json:"width"`
```
Comment (gabsn) [inline synthetics.go]: I think it should be an integer too.

**Comment by ojongerius** [comment]
Awesome @irabinovitch! @btoueg let me know when you are happy for me to merge.

**Comment by btoueg** [comment]
Hi @ojongerius, this PR is good to go, thanks!

**Comment by bkabrda** [comment]
@btoueg would you mind squashing the commits a little bit? For example, commit [1] only adds a TODO line, which is AFAICS then removed in another commit. This is relevant in terms of how this PR evolved, but not relevant to the history of the project when this gets merged. I wouldn't actually mind if this whole PR was a single commit, as it brings in a new functionality as a whole and doesn't interact much with the rest of the codebase.

[1] https://github.com/zorkian/go-datadog-api/pull/228/commits/cf85f6a9abc2c61d3af1f2737075ce39906dc70a

**Comment by btoueg** [comment]
Squash merge option should make it trivial https://github.blog/2016-04-01-squash-your-commits/ but I'll do it in case you don't have access to this functionality

**Comment by bkabrda** [comment]
@btoueg interesting, I wasn't aware that was possible :)

Either way, I don't have rights to merge (yet, should get them in couple of days) so ATM I'm just expressing my opinion. Thanks!

**Comment by btoueg** [comment]
Gonna add one commit today

**Comment by btoueg** [comment]
@bkabrda @ojongerius Is it possible to merge it ASAP?

**Comment by bkabrda** [comment]
No further comments from anyone here, the PR looks solid. I'm going to merge it. Thanks! :+1:

**Pull request merged**